### PR TITLE
fix: driver daemonset pod run failure on Windows containerd

### LINKED ISSUE #547: driver daemonset pod run failure on Windows containerd
**What happened**:

https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_azuredisk-csi-driver/546/pull-azuredisk-csi-driver-e2e-migration-windows/1308389672262569984/build-log.txt
```
print out all nodes status ...
NAME                    STATUS   ROLES    AGE   VERSION   INTERNAL-IP    EXTERNAL-IP   OS-IMAGE                         KERNEL-VERSION     CONTAINER-RUNTIME
2877k8s00000000         Ready    agent    21m   v1.18.8   10.240.0.34    <none>        Windows Server 2019 Datacenter   10.0.17763.1339    containerd://1.4.0
2877k8s00000001         Ready    agent    22m   v1.18.8   10.240.0.65    <none>        Windows Server 2019 Datacenter   10.0.17763.1339    containerd://1.4.0
k8s-master-28775873-0   Ready    master   25m   v1.18.8   10.255.255.5   <none>        Ubuntu 18.04.5 LTS               5.3.0-1035-azure   containerd://1.3.2+azure
======================================================================================
print out all kube-system namespace pods status ...
NAME                                            READY   STATUS             RESTARTS   AGE   IP             NODE                    NOMINATED NODE   READINESS GATES
azure-cni-networkmonitor-6vfsl                  1/1     Running            0          25m   10.255.255.5   k8s-master-28775873-0   <none>           <none>
azure-ip-masq-agent-8kw7j                       1/1     Running            0          25m   10.255.255.5   k8s-master-28775873-0   <none>           <none>
blobfuse-flexvol-installer-mdqrt                1/1     Running            0          25m   10.240.0.4     k8s-master-28775873-0   <none>           <none>
coredns-56db558fb9-j7vfn                        1/1     Running            0          25m   10.240.0.9     k8s-master-28775873-0   <none>           <none>
coredns-autoscaler-5c7db64899-zpf7n             1/1     Running            0          25m   10.240.0.10    k8s-master-28775873-0   <none>           <none>
csi-azuredisk-controller-7d9b85db5b-vqsgv       5/5     Running            0          17m   10.255.255.5   k8s-master-28775873-0   <none>           <none>
csi-azuredisk-node-win-jv4pk                    2/3     ImagePullBackOff   0          17m   10.240.0.77    2877k8s00000001         <none>           <none>
csi-azuredisk-node-win-kxnss                    2/3     ImagePullBackOff   0          17m   10.240.0.36    2877k8s00000000         <none>           <none>
csi-secrets-store-cg95x                         3/3     Running            0          25m   10.255.255.5   k8s-master-28775873-0   <none>           <none>
csi-secrets-store-provider-azure-4j9qf          1/1     Running            0          25m   10.240.0.13    k8s-master-28775873-0   <none>           <none>
kube-addon-manager-k8s-master-28775873-0        1/1     Running            0          25m   10.255.255.5   k8s-master-28775873-0   <none>           <none>
kube-apiserver-k8s-master-28775873-0            1/1     Running            0          24m   10.255.255.5   k8s-master-28775873-0   <none>           <none>
kube-controller-manager-k8s-master-28775873-0   1/1     Running            0          24m   10.255.255.5   k8s-master-28775873-0   <none>           <none>
kube-proxy-h2n67                                1/1     Running            0          25m   10.255.255.5   k8s-master-28775873-0   <none>           <none>
kube-scheduler-k8s-master-28775873-0            1/1     Running            0          24m   10.255.255.5   k8s-master-28775873-0   <none>           <none>
metrics-server-6756f7f765-b8s8h                 1/1     Running            0          25m   10.240.0.11    k8s-master-28775873-0   <none>           <none>
```

**What you expected to happen**:

**How to reproduce it**:

**Anything else we need to know?**:

**Environment**:
- CSI Driver version:
- Kubernetes version (use `kubectl version`):
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by abhisheksinghbaghel**
**What type of PR is this?**

**What this PR does / why we need it**:
This PR adds os version to all the images in the latest release manifest. Os version is needed for windows containerd to match the right image to download for deployment.

**Which issue(s) this PR fixes**:

Fixes #547 

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:
@andyzhangx The issue turned out to be the missing os version field in the ":latest" manifest list that we are publishing. In the recipe when we are creating "latest" manifest list, we were using the images created by buildx, but we never annotate the right os version to them.
Looks like the feature [https://github.com/docker/cli/issues/2722](url) to annotate images is implemented but not released yet.
So for now we craft two separate manifest lists and annotate the images in both of them separately. After the changes combination of container-all and push-manifest will push below images and manifest lists to registry.
TAG v1.0.0-linux-amd64
TAG v1.0.0-windows-1909-amd64
TAG v1.0.0-windows-2004-amd64
TAG v1.0.0-windows-1903-amd64
TAG v1.0.0-windows-1809-amd64
TAG v1.0.0
TAG latest

**Release note**:
We need to republish the latest manifest to the acr for latest to work with container images.

### CODE REVIEW

**Comment by andyzhangx** [comment]
per this thread, it's already fixed: https://github.com/kubernetes-sigs/azurefile-csi-driver/issues/382#issuecomment-769281163

**Comment by abhisheksinghbaghel** [comment]
There was never an issue with vx.x.x manifests because we do add the osversion to them. We need to look at latest.
Check for mcr.microsoft.com/k8s/csi/azurefile-csi:latest tag

azureuser@abhib-dev-ubuntu-18:~/go/src/sigs.k8s.io/azuredisk-csi-driver$ docker manifest inspect mcr.microsoft.com/k8s/csi/azurefile-csi:latest
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
   "manifests": [
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1165,
         "digest": "sha256:c3c285af61abe8d8aacaedc7744fc44890caedd0b18f21b1f659f413b1c1b901",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 951,
         "digest": "sha256:1c56abbee556b61d38ed80467711923d84faf6c7a6298b15ccd71b28463a73e7",
         "platform": {
            "architecture": "amd64",
            "os": "windows"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 951,
         "digest": "sha256:25544e12654e1afd997b60f351d80a94dfdd5bbbf631af5b7917bec86ddd7a2f",
         "platform": {
            "architecture": "amd64",
            "os": "windows"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 951,
         "digest": "sha256:f9b2c5019a8c06d8b9002d181962e16fec59c3c839041a6d07b06cc907222671",
         "platform": {
            "architecture": "amd64",
            "os": "windows"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 951,
         "digest": "sha256:0cdd9cc0de957047a61586252f7f6e0429b8fc901122238b6040e9cbb7d923e8",
         "platform": {
            "architecture": "amd64",
            "os": "windows"
         }
      }
   ]
}

**Comment by abhisheksinghbaghel** [comment]
@andyzhangx Same issue with the azuredisk-csi:latest

azureuser@abhib-dev-ubuntu-18:~/go/src/sigs.k8s.io/azuredisk-csi-driver/deploy$ docker manifest inspect mcr.microsoft.com/k8s/csi/azuredisk-csi:latest 
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
   "manifests": [
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 953,
         "digest": "sha256:5257238e02f4299852bf122c98306c5dadc066b065485a298168ae642ff33166",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 951,
         "digest": "sha256:11690d5d6cc8491fa8ec5c35e6c932e2524f4602f29fab5f766fbe216bab0d89",
         "platform": {
            "architecture": "amd64",
            "os": "windows"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 951,
         "digest": "sha256:4fece51c40b9eb6a87e5b0759d9a4941f1d3b1c703895a6a0bf93fb60d615e14",
         "platform": {
            "architecture": "amd64",
            "os": "windows"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 951,
         "digest": "sha256:fd79b02c61ddabd23bd97cc03c3c30f228d7a164263a7a3b8784875a4b513166",
         "platform": {
            "architecture": "amd64",
            "os": "windows"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 951,
         "digest": "sha256:b0a29fec1ec3acebb951822cf1b1890a7c3f3f040ada4fe56471cb42faf74306",
         "platform": {
            "architecture": "amd64",
            "os": "windows"
         }
      }
   ]
}

File: Makefile [hunk 1]
```diff
@@ -27,6 +27,7 @@ override IMAGE_VERSION := e2e-$(GIT_COMMIT)
 endif
 endif
 IMAGE_TAG ?= $(REGISTRY)/$(IMAGE_NAME):$(IMAGE_VERSION)
+RELEASE_TAG ?= $(REGISTRY)/$(IMAGE_NAME):$(IMAGE_VERSION)
```
Comment (andyzhangx) [inline Makefile:30]: why create a new tag var here? I think we need to fix this PR: https://github.com/kubernetes-sigs/azuredisk-csi-driver/pull/585, it has already set `os.version`, while there is sth. wrong

**Review by andyzhangx — COMMENTED** [review]
/ok-to-test

File: Makefile (hunk 1 above)
Comment (andyzhangx) [reply Makefile:30]: also should have an test to prevent such regression in the future.

**Comment by andyzhangx** [comment]
only `latest` tag does not have `os.version`:
```
# docker manifest inspect mcr.microsoft.com/k8s/csi/azuredisk-csi:v1.0.0
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
   "manifests": [
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 953,
         "digest": "sha256:5257238e02f4299852bf122c98306c5dadc066b065485a298168ae642ff33166",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 951,
         "digest": "sha256:11690d5d6cc8491fa8ec5c35e6c932e2524f4602f29fab5f766fbe216bab0d89",
         "platform": {
            "architecture": "amd64",
            "os": "windows",
            "os.version": "10.0.17763.1697"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 951,
         "digest": "sha256:4fece51c40b9eb6a87e5b0759d9a4941f1d3b1c703895a6a0bf93fb60d615e14",
         "platform": {
            "architecture": "amd64",
            "os": "windows",
            "os.version": "10.0.18362.1256"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 951,
         "digest": "sha256:fd79b02c61ddabd23bd97cc03c3c30f228d7a164263a7a3b8784875a4b513166",
         "platform": {
            "architecture": "amd64",
            "os": "windows",
            "os.version": "10.0.18363.1316"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 951,
         "digest": "sha256:b0a29fec1ec3acebb951822cf1b1890a7c3f3f040ada4fe56471cb42faf74306",
         "platform": {
            "architecture": "amd64",
            "os": "windows",
            "os.version": "10.0.19041.746"
         }
      }
   ]
}
```

File: Makefile (hunk 1 above)
Comment (andyzhangx) [reply Makefile:30]: got it, that's for latstet tag

**Comment by abhisheksinghbaghel** [comment]
@andyzhangx Do we need this anymore? If we have the new docker version. I can close this PR?

**Comment by abhisheksinghbaghel** [comment]
> @andyzhangx Do we need this anymore? If we have the new docker version. I can close this PR?

@andyzhangx ?

**Comment by andyzhangx** [comment]
> > @andyzhangx Do we need this anymore? If we have the new docker version. I can close this PR?
> 
> @andyzhangx ?

@abhisheksinghbaghel need to check, I would prefer new docker version since logic is quite clear.

**Pull request closed without merging**